src/abstractdataframe/join.jl

```diff
@@ -430,23 +345,622 @@ function Base.join(df1::AbstractDataFrame, df2::AbstractDataFrame;
     return joined
 end
 
-function Base.join(df1::AbstractDataFrame, df2::AbstractDataFrame,
-                   dfs::AbstractDataFrame...;
-                   on::Union{Symbol, AbstractVector{Symbol}} = Symbol[],
-                   kind::Symbol = :inner, makeunique::Bool=false,
-                   validate::Union{Pair{Bool, Bool}, Tuple{Bool, Bool}}=(false, false))
-    if !(kind in (:inner, :outer, :cross))
-        throw(ArgumentError("Only inner, outer, and cross joins are supported when " *
-                            "joining more than two data frames"))
-    end
-    join(join(df1, df2, on=on, kind=kind, makeunique=makeunique, validate=validate),
-         dfs..., on=on, kind=kind, makeunique=makeunique, validate=validate)
-end
+"""
+    innerjoin(df1, df2; on = Symbol[], makeunique = false,
+              indicator = nothing, validate = (false, false))
+    innerjoin(df1, df2, dfs...; on = Symbol[], makeunique = false,
+              validate = (false, false))
+
+Inner join two or more data frame objects and return a `DataFrame` containing
```

Comment:
Can you explain what an inner join is? The `join` docstring defined each kind. Same for other kinds.

Also I'm not sure whether "to inner join" is a real verb. :-) Maybe "Perform an inner join"?

Reply:
Actually, it could make sense to have a list of all possible joins pointing to other functions (like the one we currently have in `?join`). For example that's what dplyr does (since it follows the classic R tradition of documenting multiple methods in a single docstring).

Reply:
1. For each join I have added a one sentence description what it does (a bit more precise than we had before).
2. Changed to `"Perform ..."` style.
3. Added `See also` section for each docstring